nomad/eval_broker.go

```diff
@@ -934,3 +969,43 @@ func (p PendingEvaluations) Peek() *structs.Evaluation {
 	}
 	return p[n-1]
 }
+
+// PopN removes and returns the minimum N evaluations from the slice.
+func (p *PendingEvaluations) PopN(n int) []*structs.Evaluation {
+	if n > len(*p) {
+		n = len(*p)
+	}
+
+	popped := []*structs.Evaluation{}
```

Comment:
Since `n` can be significant here (1000), let's try to avoid the slice resizing.
```suggestion
	popped := make([]*structs.Evaluation, 0, n)
```

Although who knows, the Go compiler might be able to figure that out on its own.

Reply:
Agreed. Based on the testing I did for https://github.com/hashicorp/nomad/pull/15117 I think we're probably going to want to crank the slice of the max size down quite a bit as well.